agent/config.go

```diff
@@ -0,0 +1,82 @@
+package core
+
+import (
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/appcelerator/amp/pkg/config"
+)
+
+// AgentConfig configuration parameters
+type AgentConfig struct {
+	dockerEngine string
+	apiPort      string
+	period       int
+	natsURL      string
+	clientID     string
+	clusterID    string
+}
+
+var conf AgentConfig
+
+// Update conf instance with default value and environment variables
+func (cfg *AgentConfig) init(version, build string) {
+	cfg.setDefault()
+	cfg.loadConfigUsingEnvVariable()
+	cfg.displayConfig(version, build)
+}
+
+// Set default value of configuration
+func (cfg *AgentConfig) setDefault() {
+	cfg.dockerEngine = amp.DockerDefaultURL
+	cfg.natsURL = amp.NatsDefaultURL
+	cfg.apiPort = "3000"
+	cfg.period = 3
+	cfg.clientID = "agent-" + os.Getenv("HOSTNAME")
+	cfg.clusterID = amp.NatsClusterID
+}
+
+// Update config with env variables
+func (cfg *AgentConfig) loadConfigUsingEnvVariable() {
+	cfg.dockerEngine = getStringParameter("DOCKER", cfg.dockerEngine)
+	cfg.natsURL = getStringParameter("NATS_URL", cfg.natsURL)
+	cfg.apiPort = getStringParameter("API_PORT", cfg.apiPort)
+	cfg.period = getIntParameter("PERIOD", cfg.period)
+	cfg.clientID = getStringParameter("CLIENTID", cfg.clientID)
+	cfg.clusterID = getStringParameter("CLIENTID", cfg.clusterID)
+}
+
+// Display amp-pilot configuration
+func (cfg *AgentConfig) displayConfig(version, build string) {
+	log.Printf("agent version: %s, build: %s", version, build)
+	log.Println("----------------------------------------------------------------------------")
+	log.Println("Configuration:")
+	log.Printf("Docker-engine: %s\n", conf.dockerEngine)
+	log.Printf("Nats URL: %s\n", conf.natsURL)
+	log.Printf("ClientId: %s\n", conf.clientID)
+	log.Printf("ClusterId: %s\n", conf.clusterID)
+	log.Println("----------------------------------------------------------------------------")
+}
+
+// Return env variable value, if empty return default value
+func getStringParameter(envVariableName string, def string) string {
+	value := os.Getenv(envVariableName)
```

Comment:
LookupEnv would be better than Getenv, to make the difference between unset and empty values.